### ISSUE TITLE
Memory Leaks

### ISSUE THREAD

**Issue by bill-foreflight** [microsoft/plcrashreporter#171]
### **Description**

Calling `generateLiveReport` multiple times will trigger a crash. Instruments will not report the memory leak/allocations but the log will print:

`malloc: can't allocate region`

We used PLCrashReporter for a few years, and we first started having this problem after we upgraded to version 1.8.1 a few months ago. Prior to that, we used the same version since July 2019 without known memory issues (not sure what that version number was).

### **Repro Steps**

Call `generateLiveReport` repeatedly. The issue occurs in Objective-c, (even with autoreleasepools), while generating live reports in a loop. It also happens in Swift with the below example that calls repeatedly via GCD, which should clean up memory on recursive calls.
```
import CrashReporter

class ViewController: UIViewController {
   
    let reporter = PLCrashReporter(configuration: PLCrashReporterConfig.defaultConfiguration())!
    var callCount = 0
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        test()
    }
    
    func test() {
        
        DispatchQueue.global(qos: .userInitiated).async {
            self.reporter.generateLiveReport()
            self.callCount += 1
            //print("Called \(self.callCount) times")
            
            DispatchQueue.main.async {
                self.test()
            }
        }
    }
}
```

### **Details**

1. Which SDK version are you using? 1.8.1 for iOS
2. Which OS version did you experience the issue on? iOS 14.4
3. Which CocoaPods/Carthage/Xcode version are you using? Xcode 12.4
4. What device version did you see this error on? Were you using an emulator or a physical device? Mulitple devices but iPad Pro 11"
5. What language are you using? Issue occurs with both.
    - [ x ] Objective C
    - [ x ] Swift
6. What third party libraries are you using? Sample code above uses no 3rd party libraries.

**Comment by MatkovIvan** [comment]
Hi, thanks for the report, we'll take a look.
For now, I can only say that the related change is only community contribution #93 that introduces usage memory mapping options to avoid OOM in similar to your case.

cc @adubovkin in case if you are still around.

**Comment by MatkovIvan** [comment]
Well, after closer look, it is not related to recent changes. Feel free to try my changes that fixes this - #172

**Comment by MatkovIvan** [comment]
Fixed by #172
